ggshield/output/text/text_output.py

```diff
@@ -52,7 +52,15 @@ def process_scan(self, scan: ScanCollection, top: bool = True) -> Tuple[str, int
                 scan_buf.write(self.process_result(result))
         else:
             if self.verbose:
-                scan_buf.write(no_leak_message())
+                has_results = False
+                if scan.scans:
+                    for sub_scan in scan.scans:
+                        if sub_scan.results:
+                            has_results = True
+                            break
```

Comment:
I think this can be simplified to:
```suggestion
                    has_results = any(x.results for x in scan.scans)
```